### PR TITLE
Fix unicode build on Windows

### PR DESCRIPTION
**Pull request by garbear**
As title says. Thanks to @Paxxi for the hint.

### CODE REVIEW

**Comment by garbear** [comment]
@Paxxi has jenkins been changed over to unicode yet? This fails to build on windows jenkins

File: src/api/directinput/JoystickInterfaceDirectInput.cpp [hunk 1]
```diff
@@ -142,7 +143,9 @@ BOOL CALLBACK CJoystickInterfaceDirectInput::EnumJoysticksCallback(const DIDEVIC
     return DIENUM_CONTINUE;
   }
 
-  const std::string strName = pdidInstance->tszProductName ? pdidInstance->tszProductName : "";
+  std::string strName;
+  if (pdidInstance->tszProductName)
+    strName = KODI::PLATFORM::WINDOWS::FromW(std::wstring(pdidInstance->tszProductName));
```
Comment (Rechi) [inline src/api/directinput/JoystickInterfaceDirectInput.cpp]: `char[]` can't be passed to the `std::wstring` constructor

Comment (garbear) [inline src/api/directinput/JoystickInterfaceDirectInput.cpp]: when unicode is enabled, the type is WCHAR[]. is there a #define i can use to check this?

Comment (Paxxi) [inline src/api/directinput/JoystickInterfaceDirectInput.cpp]: According to https://msdn.microsoft.com/en-us/library/windows/desktop/microsoft.directx_sdk.reference.dideviceinstance(v=vs.85).aspnet it should be a tchar which evaluates to a wchar_t so not sure what's wrong but should be solved by defining UNICODE and _UNICODE before including windows.h should do the trick

File: src/api/xinput/XInputDLL.cpp [hunk 2]
```diff
@@ -47,16 +48,16 @@ bool CXInputDLL::Load(void)
 
   m_strVersion = "1.4";
   dsyslog("Attempting to load XInput1_4.dll...");
-  m_dll = LoadLibrary("XInput1_4.dll");  // 1.4 Ships with Windows 8
+  m_dll = LoadLibrary(KODI::PLATFORM::WINDOWS::ToW("XInput1_4.dll").c_str());  // 1.4 Ships with Windows 8
```
Comment (Paxxi) [inline src/api/xinput/XInputDLL.cpp]: Easier to just add an L before the "", no conversion necessary. If you want compatibility with both versions you can use the T() macro instead

**Comment by garbear** [comment]
Updated with suggestions. Is this OK now?

**Comment by garbear** [comment]
@Paxxi can you review before I merge?

**Pull request merged**